Assets/MRTK/Providers/WindowsMixedReality/XR2018/WindowsMixedRealityDeviceManager.cs

```diff
@@ -688,13 +741,27 @@ private BaseWindowsMixedRealitySource GetOrAddController(InteractionSource inter
                 }
                 else if (interactionSource.kind == InteractionSourceKind.Controller)
                 {
-                    detectedController = new WindowsMixedRealityController(TrackingState.NotTracked, controllingHand, inputSource);
-                    if (!detectedController.Enabled)
+#if HP_CONTROLLER_ENABLED
+                    if (isHPController)
                     {
-                        // Controller failed to be setup correctly.
-                        // Return null so we don't raise the source detected.
-                        return null;
+                        lock (trackedMotionControllerStates)
+                        {
+                            // Add the controller as a HP Motion Controller
+                            HPMotionController hpController = new HPMotionController(TrackingState.NotTracked, controllingHand, inputSource);
+                            if (trackedMotionControllerStates.ContainsKey(controllerId))
+                            {
+                                hpController.MotionControllerState = trackedMotionControllerStates[controllerId];
+                            }
+                            detectedController = hpController;
+                        }
                     }
+                    else
+                    {
+                        detectedController = new WindowsMixedRealityController(TrackingState.NotTracked, controllingHand, inputSource);
+                    }
+#else
+                    detectedController = new WindowsMixedRealityController(TrackingState.NotTracked, controllingHand, inputSource);
+#endif
```

Comment:
Similar question here. I don't think the `HPMotionController` class fails to run if `HP_CONTROLLER_ENABLED` isn't true (at least, it looks like it successfully falls back to the base case? we should make sure it still works!). Can we just always spin up an `HPMotionController` is `isHPController` and a `WindowsMixedRealityController` otherwise? I assume the
```c#
if (trackedMotionControllerStates.ContainsKey(controllerId))
{
    hpController.MotionControllerState = trackedMotionControllerStates[controllerId];
}
```
block will need to stay in an `#if`, but it'll be scoped down a bit